R/release.R

```diff
@@ -122,13 +129,32 @@ release_checklist <- function(version, on_cran) {
   )
 }
 
-release_extra <- function(env = NULL) {
-  if (is.null(env)) {
-    env <- tryCatch(
-      pkg_env(project_name()),
-      error = function(e) emptyenv()
-    )
+release_revdepcheck <- function(on_cran = TRUE, is_rstudio_pkg = TRUE, env = NULL) {
+  if (!on_cran) {
+    return()
+  }
+
+  env <- env %||% safe_pkg_env()
+  if (env_has(env, "release_extra_revdeps")) {
+    extra <- env$release_extra_revdeps()
+  } else {
+    extra <- character()
+  }
+
+  if (is_rstudio_pkg) {
+    if (length(extra) > 1) {
+      extra_code <- paste0(deparse(extra), collapse = "")
```

Comment:
I had to run this locally to figure out what it did 😅

Reply:
Can it just be `deparse(extra)`?

``` r
extra <- letters[1:3]
identical(deparse(extra), paste0(deparse(extra), collapse = ""))
#> [1] TRUE
```

Reply:
Probably would be fine here, but it's safer to always collapse into a single string:

``` r
extra <- paste0("longpackagename", letters[1:10])
length(deparse(extra))
#> [1] 3
```

<sup>Created on 2023-01-24 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>